nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -79,6 +80,59 @@ in {
       default = "/var/lib/nextcloud";
       description = "Storage path of nextcloud.";
     };
+    datadir = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        Data storage path of nextcloud.  Will be <xref linkend="opt-services.nextcloud.home" /> by default.
+        This folder will be populated with a config.php and data folder which contains the state of the instance (excl the database).";
+      '';
+      example = "/mnt/nextcloud-file";
+    };
+    extraApps = mkOption {
+      type = types.attrsOf types.package;
```

Comment:
In the future, it would be nice to provide a set of tested Nextcloud apps. The user could then choose to enable a set of prepackaged Nextcloud apps or add its own. I'm not sure the current type would be convenient to choose an app amongst a predefined list.

A type such as `extraApps = types.listOf types.package` would then allow us to create the following type in the fututre `types.listOf (types.Or types.package (types.enum ['calendar' 'contact']))`.

Currently, the attribute set key is only used to define the app directory name. What about using the `id` value found in the `appinfo/info.xml` file instead?